### PR TITLE
Install parallel gzip utility

### PR DESCRIPTION
**Pull request by am11**
Related https://github.com/dotnet/arcade/pull/15038.

### CODE REVIEW

**Comment by am11** [comment]
cc @akoeplinger, installed in all files which are installing `unzip` utility.

File: src/azurelinux/3.0/net9.0/android/Dockerfile [hunk 1]
```diff
@@ -5,9 +5,10 @@ RUN tdnf update -y \
     && tdnf install -y \
         # Common dependencies
         binutils \
+        pigz \
```
Comment (akoeplinger) [inline src/azurelinux/3.0/net9.0/android/Dockerfile:8]: I think I'd prefer installing it in the base image instead

File: src/debian/11/helix/amd64/Dockerfile [hunk 2]
```diff
@@ -23,6 +23,7 @@ RUN apt-get update && \
         llvm \
         locales \
         locales-all \
+        pigz \
```
Comment (akoeplinger) [inline src/debian/11/helix/amd64/Dockerfile:26]: we don't need this in the helix images, those don't create the .tar.gz archives

File: src/azurelinux/3.0/net9.0/android/Dockerfile (hunk 1 above)
Comment (am11) [reply src/azurelinux/3.0/net9.0/android/Dockerfile:8]: I installed it next to unzip, maybe that's not the right criteria?

Comment (akoeplinger) [reply src/azurelinux/3.0/net9.0/android/Dockerfile:8]: yeah we should really only need it in a very small subset of images

Comment (am11) [reply src/azurelinux/3.0/net9.0/android/Dockerfile]: Yup, updated the root ones, without touching helix stuff.

File: src/azurelinux/3.0/net9.0/crossdeps-builder/amd64/Dockerfile [hunk 3]
```diff
@@ -4,6 +4,7 @@ RUN tdnf install -y \
         # Common utilities
         ca-certificates \
         git \
+        pigz \
```
Comment (jkoritzinsky) [inline src/azurelinux/3.0/net9.0/crossdeps-builder/amd64/Dockerfile:7]: We don't need it in the crossdeps-builder images as we only use these images to build components that are included in other images.

File: src/azurelinux/3.0/net9.0/fpm/amd64/Dockerfile [hunk 4]
```diff
@@ -5,6 +5,7 @@ RUN tdnf install -y \
         ca-certificates \
         git \
         icu \
+        pigz \
```
Comment (jkoritzinsky) [inline src/azurelinux/3.0/net9.0/fpm/amd64/Dockerfile:8]: We don't need it in this image as we only use this image to run fpm to make rpms.

File: src/centos/stream9/amd64/Dockerfile [hunk 5]
```diff
@@ -43,6 +43,7 @@ RUN dnf install --setopt tsflags=nodocs --refresh -y \
         ncurses-devel \
         numactl-devel \
         openssl-devel \
+        pigz \
```
Comment (jkoritzinsky) [inline src/centos/stream9/amd64/Dockerfile:46]: Is pigz in the base repository for Centos Stream 9? I'd like to keep this image as similar as possible to how our partners at RedHat would build dotnet from source.

File: src/debian/11/opt/arm64v8/Dockerfile [hunk 6]
```diff
@@ -25,6 +25,7 @@ RUN apt-get update \
             llvm-9 \
             locales \
             make \
+            pigz \
```
Comment (jkoritzinsky) [inline src/debian/11/opt/arm64v8/Dockerfile:28]: We use this image to generate profile data, we don't build on it. So we don't need to include this package here.

File: src/opensuse/15.4/helix/amd64/Dockerfile [hunk 7]
```diff
@@ -26,6 +26,7 @@ RUN zypper ref && \
         lldb-devel \
         llvm \
         lttng-ust-devel \
+        pigz \
```
Comment (jkoritzinsky) [inline src/opensuse/15.4/helix/amd64/Dockerfile:29]: This is a helix image, so we don't need this here.

File: src/azurelinux/3.0/net9.0/crossdeps-builder/amd64/Dockerfile (hunk 3 above)
Comment (am11) [reply src/azurelinux/3.0/net9.0/crossdeps-builder/amd64/Dockerfile:7]:
```suggestion
```

File: src/azurelinux/3.0/net9.0/fpm/amd64/Dockerfile (hunk 4 above)
Comment (am11) [reply src/azurelinux/3.0/net9.0/fpm/amd64/Dockerfile:8]:
```suggestion
```

File: src/centos/stream9/amd64/Dockerfile (hunk 5 above)
Comment (am11) [reply src/centos/stream9/amd64/Dockerfile:46]: Yes, it is available there. This only improves the speed up our tar usage when available; perf numbers are in the linked issue.

File: src/debian/11/opt/arm64v8/Dockerfile (hunk 6 above)
Comment (am11) [reply src/debian/11/opt/arm64v8/Dockerfile:28]:
```suggestion
```

File: src/opensuse/15.4/helix/amd64/Dockerfile (hunk 7 above)
Comment (am11) [reply src/opensuse/15.4/helix/amd64/Dockerfile:29]:
```suggestion
```

File: src/centos/stream9/amd64/Dockerfile (hunk 5 above)
Comment (am11) [reply src/centos/stream9/amd64/Dockerfile:46]: https://github.com/dotnet/arcade/pull/15038#discussion_r1743480848

**Review by jkoritzinsky — APPROVED** [review]

File: src/centos/stream9/amd64/Dockerfile (hunk 5 above)
Comment (am11) [reply src/centos/stream9/amd64/Dockerfile:46]: Looks like CentOS found it in baseos repo:

image

Comment (am11) [reply src/centos/stream9/amd64/Dockerfile:46]: cc @omajid FYI

**Comment by am11** [comment]
@akoeplinger, I've addressed the feedback. Could you take another look?

**Comment by am11** [comment]
Reopened https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/1202 to get haiku change flow in from arcade.

**Pull request closed without merging**